docs/rfcs/0003-etag-changes.md

```diff
@@ -24,54 +24,153 @@ While the current Etag implementation allows for easily adding and removing Etag
+
+  - This file hash is pre-generated and would be easy to retreive
 - This has issues with storage collections, as these aren't stored in S3 - we could use a different system for generating this hash, but then that means there is a different code journey in this specific scenario, which has caused issues with other parts of the presentation API
-- The S3 document updates at some point in the future when there are Assets attached to a manifest (i.e.: staging to live) - dealing with this would add complexity
+- The S3 document updates at some point in the future when there are Assets attached to a manifest (i.e.: staging to live) - dealing with this would add complexity.  Though it would likely be easier to retrieve from the 
+- There are some operations (like changing the hierarchy or slug) that won't result in changes to S3, meaning the hash won't be updated.  This is incorrect, so instead the hash would need to be updated 
 
 #### Generated Etag
 
-The final option is to not use the request/response to generate an Etag, but instead to use something like the id generator to generate a random identifier for an etag.  This could avoid issues with differences in request/response but create potential problems around collisions.  However, this might not be an issue as it's unlikely to occur and even if it doid, there would be a very low chance of using these commmon identifiers to update different resources
+The final option is to not use the request/response to generate an Etag, but instead to use something like the id generator to generate a random identifier for an etag.  This could avoid issues with differences in request/response but create potential problems around collisions.  However, this might not be an issue as it's unlikely to occur and even if it did, there would be a very low chance of using these commmon identifiers to update different resources.
+
+Geenrating the Etag could also be combined with data to indicate versioning (i.e. create has v1 encoded, first update v2 etc.) but care must be taken to make sure that these Etags look truly random as it's possible users begin ascribing meaning to the Etag values if they are too similar or seemingly follow a convention.
+
+## Recomendations
 
-### Other considerations
+Based on the information discussed above, these are what the new Etags process should do:
+
+- The Etag should be stored in the database under a separate `etag` table
+- While the distributed cache would be useful, as an initial implementation it would be ok to run using the direct database calls.  However, this should be monitored and if it seems that the database begins to become resource starved, then the distributed cache should be imlemented.  However, in the intervening time between deciding the cache is needed and deploying the implementation, additional resources can be dedicated to the [RDS instance](https://aws.amazon.com/rds/instance-types/) to mitigate issues with resource starvation.
```

Comment:
One option could be to split logic for read+write requests, assuming GET requests _vastly_ outnumber write requests we can optimise for that.

Read requests:
* On receipt of GET w/ etag, get value from cache (read-through cache so may originate from DB)
* If we match, return 304 (memory-lookup only, inexpensive)
* else, read content from DB + s3 and return w/ 200. On way out refresh cache with DB etag if it differs to ensure fresh.
  * If the given etag was for a later version than we had in-memory all we've done is fetch once from DB so minimal extra work.

Write requests:
* On receipt of PUT/POST w/ etag, get value from _db_ - ensures we're always look at the latest one.
* If eTag matches or not allow or not. Main thing is we got this fresh.